src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +154,555 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app;
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg)[am::strings::msg_params][am::strings::info] = "info1";
+
+  Event event(hmi_apis::FunctionID::UI_SetAppIcon);
+  event.set_smart_object(*msg);
+
+  req_vr->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, Run_AllCorrectData_SendHMIRequest) {
+  MessageSharedPtr msg = CreateFullParamsUISO();
+  const std::string sync_file_name = "sync_file";
+  (*msg)[am::strings::msg_params][am::strings::sync_file_name] = sync_file_name;
+
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillRepeatedly(Return(mock_app));
+
+  MockApplicationManagerSettings mock_app_mngr_settings;
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_settings));
+
+  const std::string app_storeage_folder = "./storage";
+  EXPECT_CALL(mock_app_mngr_settings, app_storage_folder())
+      .WillRepeatedly(ReturnRef(app_storeage_folder));
+
+  const std::string app_folder = "Test_app";
+  EXPECT_CALL((*mock_app), folder_name()).WillRepeatedly(Return(app_folder));
+
+  MockProtocolHandler mock_app_mngr_protocol_hendler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_protocol_hendler));
+
+  MockProtocolHandlerSettings mock_app_mngr_protocol_hendler_settings;
+  EXPECT_CALL(mock_app_mngr_protocol_hendler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_protocol_hendler_settings));
+
+  const protocol_handler::MajorProtocolVersion mock_app_mngr_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_app_mngr_protocol_hendler_settings,
+              max_supported_protocol_version())
+      .WillRepeatedly(Return(mock_app_mngr_protocol_version));
+
+  const std::string icon_folder = app_storeage_folder + "/" + app_folder;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder())
+      .WillRepeatedly(ReturnRef(icon_folder));
+
+  const uint32_t storage_max_size = 10;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  const std::string app_policy_id = "test_icon";
+  EXPECT_CALL((*mock_app), policy_app_id())
+      .WillRepeatedly(Return(app_policy_id));
+
+  const uint32_t app_id = 152;
+  EXPECT_CALL((*mock_app), app_id()).WillRepeatedly(Return(app_id));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  std::string directory = app_storeage_folder + "/" + app_folder;
```

Comment:
@BSolonenko const?

Reply:
@AByzhynar Added in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/d3e063fe7994d5ed83665c9492a000bb0dc47552